### ISSUE TITLE
High CPU usage in development environment.

### ISSUE THREAD

**Issue by liaosong** [angular/angular-cli#10716]
Hi guys, I started up a new project with below version. The high CPU usage is showing, Even in idle time. 
### Versions
```
Angular CLI: 1.7.4
Node: 8.11.1
OS: darwin x64
Angular: 6.0.0

```

### Observed behavior
![image](https://user-images.githubusercontent.com/3927282/39736699-e0c5ad98-52b3-11e8-8682-f6af8257a159.png)

**Comment by phl3x0r** [comment]
I don't think this is related to angular CLI. Have you tried figuring out what is causing the CPU usage?
If you use VSCode (which many do), it may very well be caused by an extension. You can test this out by disabling all your extensions and see if it gets better, then enable them one at a time to find the culprit.

**Comment by liaosong** [comment]
@phl3x0r I closed all the IDE, And started project in terminal, The CPU usage is still high.

**Comment by liaosong** [comment]
I found the `ng` process is NO.1 of the `hard disk` usage.

**Comment by mmmichl** [comment]
Do you run mac? Check if you have `fsevents` install, if not add it: `npm i --no-save fsevents` and see if the high cpu usage goes away.

**Comment by smlombardi** [comment]
I noticed today after updating to Angular 6 my fans are going nuts on my 2017 MBP, 10.13.4, so I checked and NG averages 166%, maxing at 268% CPU.

<img width="387" alt="screen shot 2018-05-09 at 3 52 28 pm" src="https://user-images.githubusercontent.com/835148/39836394-39825a10-53a1-11e8-98da-0fb3b599d861.png">

**Comment by liaosong** [comment]
@smlombardi My MBP version is `MacBook Pro (13-inch, 2017, Two Thunderbolt 3 ports)`, When I upgrade `angular-cli` to 6.0.0. The CPU high usage goes away.

**Comment by smlombardi** [comment]
@liaosong Not for me, 6.0.0 still spikes my CPU — fans going nuts the whole time server is running

**Comment by liaosong** [comment]
@smlombardi Remove your `node_modules` and reinstall it. I change the version to `1.7.4`, It works well.

**Comment by smlombardi** [comment]
@liaosong  That helped.  Stuck with 6.0.0. Thanks. 👍🏻

**Comment by tucq88** [comment]
> Do you run mac? Check if you have `fsevents` install, if not add it: `npm i --no-save fsevents` and see if the high cpu usage goes away.

You saved me @mmmichl . I have been struggled with this `high CPU` problem for nearly a year, since Angular 4 to now 6, and I blame that my macbook pro full options mid 2015 was out-of-date, can't be used for web development anymore. 
Could you please explain why and how fsevents was that magical ? Or just ref links is enough. Thanks and wish you a nice day!

**Comment by mmmichl** [comment]
@tucq88 I cannot give you details on the reasons, maybe the package description will help you https://www.npmjs.com/package/fsevents

Just be cautious to _not add_ it to `package.json`! Ohterwise the installation will fail on Linux or Windows.

**Comment by xiaohuilam** [comment]
![image](https://user-images.githubusercontent.com/6964962/54656475-b9c09500-4b00-11e9-8ef8-9de05a47c290.png)

angular makes me anger

**Comment by radovansurlak** [comment]
Hey guys, I started using https://github.com/fuyu/AppPolice to limit the CPU load of `ng serve`.
The load was making my Mac really slow, this one allows you to limit the CPU usage.

It doesn't solve the cause, but at least I can work now :D
